backends/bevy_picking_ui/src/lib.rs

```diff
@@ -93,16 +95,27 @@ pub fn ui_picking(
         // consider the topmost one rendering UI in this window.
         let mut ui_cameras: Vec<_> = cameras
             .iter()
-            .filter(|(_entity, camera, _)| {
+            .filter(|(_, camera)|
                 camera.is_active
                     && camera.target.normalize(Some(window_entity)).unwrap() == location.target
+            )
+            .filter(move |(_entity, camera)| {
+                let matching_root_node = root_nodes
+                    .iter()
+                    .find(|(_, visibility, root_node_target_camera_marker_option)| {
+                        root_node_target_camera_marker_option
+                            .and_then(|marker| cameras.get(marker.entity()).ok())
+                            .map_or(false, |(_, target_camera)| target_camera == camera)
+                    });
+                matching_root_node.map_or(true, |(_, visibility, _)| {
+                    matches!(visibility, Visibility::Visible | Visibility::Inherited)
```

Comment:
I assumed `Visibility::Inherited` should be treated like `Visible` since NodeBundles spawned at the root default to `Visibility::Inherited`